src/plots/cartesian/axes.js

```diff
@@ -1684,7 +1684,8 @@ axes.doTicks = function(gd, axid, skipTitle) {
                     fontSize*(offsetBase + (ax.showticklabels ? 1.5 : 0.5)));
 
             if(ax.rangeslider && ax.rangeslider.visible && ax._boundingBox) {
-                y += gs.h * ax.rangeslider.thickness + ax._boundingBox.height;
+                y += (fullLayout.height - fullLayout.margin.b - fullLayout.margin.t) *
+                    ax.rangeslider.thickness + ax._boundingBox.height;
```

Comment:
OK @mdtusz I'm not sure what's going on here and why what I put in has a different effect, but I put back in the logic you had. I'm going to make a separate issue out of it (or two perhaps) because rangesliders don't currently resize correctly when something auto-increases the margin, but also dragging legends (when we have `context = {editable: true}`) seems to be broken on master.